redbot/cogs/downloader/downloader.py

```diff
@@ -1214,23 +1270,35 @@ async def _cog_list(self, ctx: commands.Context, repo: Repo) -> None:
 
         - `<repo>` The repo to list cogs from.
         """
-        installed = await self.installed_cogs()
-        installed_str = ""
-        if installed:
-            installed_str = _("Installed Cogs:\n") + "\n".join(
-                [
-                    "- {}{}".format(i.name, ": {}".format(i.short) if i.short else "")
-                    for i in installed
-                    if i.repo_name == repo.name
-                ]
-            )
-        cogs = _("Available Cogs:\n") + "\n".join(
-            [
-                "+ {}: {}".format(cog.name, cog.short or "")
-                for cog in repo.available_cogs
-                if not (cog.hidden or cog in installed)
-            ]
+        all_installed_cogs = await self.installed_cogs()
+        installed_cogs_in_repo = [cog for cog in all_installed_cogs if cog.repo_name == repo.name]
+        installed_str = "\n".join(
+            "- {}{}".format(i.name, ": {}".format(i.short) if i.short else "")
+            for i in installed_cogs_in_repo
         )
+
+        if not installed_str:
+            installed_str = _("No cogs installed.")
+        elif len(installed_cogs_in_repo) > 1:
+            installed_str = _("Installed Cogs:\n{text}").format(text=installed_str)
+        else:
+            installed_str = _("Installed Cog:\n{text}").format(text=installed_str)
+        available_str = "\n".join(
+            "+ {}{}".format(cog.name, ": {}".format(cog.short) if cog.short else "")
+            for cog in repo.available_cogs
+            if not (cog.hidden or cog in installed_cogs_in_repo)
+        )
+
+        available_cogs = sum(
+            not cog.hidden and cog not in installed_cogs_in_repo for cog in repo.available_cogs
+        )
+
+        if not available_str:
+            cogs = _("No cogs available.")
```

Comment:
Contrary to what I said for "Installed Cogs" section, I think that *here* it would actually make sense to show this information. However, I think it would be much better if it were:
```suggestion
        if not available_str:
            cogs = _("Available Cogs:\nNo cogs are available.")
```